internal/datastore/crdb/tx.go

```diff
@@ -161,7 +161,8 @@ func executeWithRetries(ctx context.Context, currentTx pgx.Tx, fn transactionFn,
 func resetExecution(ctx context.Context, conn conn, tx pgx.Tx, txOptions pgx.TxOptions) (newTx pgx.Tx, err error) {
 	if tx != nil {
 		err = tx.Rollback(ctx)
-		if err != nil {
+		// May return ErrTxClosed if tx is already closed and can be ignored since we're resetting
+		if err != nil && !errors.Is(err, pgx.ErrTxClosed) {
```

Comment:
when would this already be closed? if we're calling `resetExecution` we shouldn't have called rollback or commit before, right?

Reply:
Perhaps here? Initial error is retriable, rollback, rollback returns a resettable error https://github.com/authzed/spicedb/pull/298/files#diff-f86db8a3a74f2b860b9f814eeb6ca84aea022526eaefedf57120a5f1a32ffe80R148

Rollback is happening in the failure that inspired this change: https://github.com/authzed/spicedb/runs/5206162023?check_suite_focus=true#step:4:35313

Reply:
Retriable errors should "release" the savepoint, not actually rollback/close the transaction though. This makes me think there's a bug where we can close the transaction twice

Reply:
So reading through the integration test error again:
```
{"level":"info","module":"pgx","args":[],"commandTag":"U0FWRVBPSU5U","sql":"SAVEPOINT cockroach_restart","time":0.471605,"time":"2022-02-15T19:45:28Z","message":"Exec"}
{"level":"info","module":"pgx","args":[],"commandTag":"Uk9MTEJBQ0s=","sql":"ROLLBACK TO SAVEPOINT cockroach_restart","time":1.245013,"time":"2022-02-15T19:45:28Z","message":"Exec"}
{"level":"error","error":":  (SQLSTATE 40003)","time":"2022-02-15T19:45:28Z","message":"resettable error, reconnecting with new transaction"}
{"level":"info","module":"pgx","args":[],"commandTag":"Uk9MTEJBQ0s=","sql":"rollback","time":0.197202,"time":"2022-02-15T19:45:28Z","message":"Exec"}
{"level":"info","module":"pgx","time":"2022-02-15T19:45:28Z","message":"closed connection"}
```

I believe it's actually just the retry/reset loop completing so no last reset is actually attempted. I've updated the log statements to be more accurate but that leaves the question of ignoring `ErrTxClosed`. Should we leave it in since we're resetting the tx anyway or remove it to be explicit that we expect transactions to only be closed by this code once?

Reply:
@samkim looking at the error you linked, I think it was just a test flake. I saw the same one a while ago and reported it: https://github.com/authzed/spicedb/issues/209

I think it's fine to revert this line, I think it was fine as it was

Reply:
Reverted the ErrTxClosed ignore